File: src/dataset/go/pangram_Exercism_go_thinkverse/openAI/ChatGPT4_pangram_v4.go
package pangram

func IsPangram(input string) bool {
	var mask uint32
	for i := 0; i < len(input); i++ {
		c := input[i]
		if c >= 'A' && c <= 'Z' {
			c += 'a' - 'A'
		}
		if c >= 'a' && c <= 'z' {
			mask |= 1 << (c - 'a')
			if mask == (1<<26)-1 {
				return true
			}
		}
	}
	return mask == (1<<26)-1
}

File: src/dataset/cpp/pangram_Exercism_johnngugi/openAI/ChatGPT4_pangram_v4.cpp
#include <string>
#include "pangram.h"

namespace pangram {

    bool is_pangram(std::string text) {
        unsigned int mask = 0;
        for (char c : text) {
            if (isalpha(c)) {
                mask |= 1u << (tolower(c) - 'a');
                if (mask == 0x3FFFFFF) return true;
            }
        }
        return mask == 0x3FFFFFF;
    }
}